.github/workflows/buildAndTest.yaml

```diff
@@ -126,3 +126,46 @@ jobs:
         with:
           name: Test logs - ${{ matrix.it}}
           path: "**/target/failsafe-reports/*"
+
+  linkageCheck:
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: install
+        # install before running Linkage Checker
+        run: |
+          ./mvnw \
+             --batch-mode \
+             --show-version \
+             --threads 1.5C \
+             --define maven.test.skip=true \
+             --define skip.surefire.tests=true \
+             --define skipITs=true \
+             --define maven.javadoc.skip=true \
+             --errors \
+             install
+      - name: run linkage checker
+        run: |
+          ./mvnw \
+            --activate-profiles linkage-check \
+            --batch-mode \
+            --show-version \
+            --threads 1.5C \
+            --define maven.test.skip=true \
+            --define skip.surefire.tests=true \
+            --define skipITs=true \
```

Comment:
The other build and test job is also doing an mvn verify, but actually runs the integration tests. 

a) does the link checker only run in the verify stage, and if so
b) we currently run ITs as a presubmit for internal folks and on merge to master for external contributions. Is that sufficient enough to just roll with it in the existing 'build and test' job?

Reply:
a) Yes, it only runs verify phase.

b) This can be added to the existing build and test. But that would add additional 13 minutes ([the previous run](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/56/checks?check_run_id=1133866122) took 13 mins) to mark a PR as green. I think you want to run this Linkage Checker in parallel to existing GitHub Actions.

Reply:
b) Yeah, let's definitely run this one in parallel.